com.unity.cinemachine/Editor/Upgrader/UpgradeObjectToCm3.cs

```diff
@@ -4,6 +4,7 @@
 using System;
 using System.Linq;
 using Cinemachine.Utility;
+using NUnit.Framework;
```

Comment:
NUnit?  no unit tests here.  We should not have this dependancy

Reply:
I am using assert in the switch statement to throw an error if we ever have a new path type -> unlikely. I could remove the assert or add a throw new NotImplementedException.

Reply:
The user may have a special path. So maybe throwing the not implemented exception is good. Or we could just ignore those paths.

Reply:
Tested. If user has a custom CinemachinePathBase, then we add a Spline (so that the upgraded trackedDolly can reference it) on it and display error in console saying that their custom thing was not updated. The gameobject also gets the CinemachineDoNotUpgrade tag.

Is this the desired behaviour? I am not sure if the `CinemachineDoNotUpgrade` tag is necessary.

Reply:
I guess it's good enough for now